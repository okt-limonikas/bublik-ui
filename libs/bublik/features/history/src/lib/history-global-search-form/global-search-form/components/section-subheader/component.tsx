/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { FC } from 'react';

export interface FormSectionSubheaderProps {
	name: string;
}

export const FormSectionSubheader: FC<FormSectionSubheaderProps> = ({
	name
}) => {
	return (
		<div className="mb-2">
			<span className="text-[0.6875rem] leading-[0.875rem]">{name}</span>
		</div>
	);
};
