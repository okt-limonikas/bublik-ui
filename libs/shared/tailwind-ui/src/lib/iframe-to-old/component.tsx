/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, FC } from 'react';

import { cn } from '../utils';

export type IframeToOldBublikProps = ComponentPropsWithRef<'iframe'>;

export const IframeToOldBublik: FC<IframeToOldBublikProps> = (props) => {
	return (
		<iframe
			className={cn('w-full h-full border-none', props.className)}
			title="Old Bublik"
			{...props}
		/>
	);
};
