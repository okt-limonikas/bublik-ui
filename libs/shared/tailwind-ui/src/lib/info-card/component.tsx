/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, forwardRef } from 'react';

export interface InfoCardProps extends ComponentPropsWithRef<'div'> {
	header: string;
	message: string;
}

export const InfoCard = forwardRef<HTMLDivElement, InfoCardProps>(
	(props, ref) => {
		const { header, message } = props;

		return (
			<div
				className="flex items-center justify-center w-full h-screen"
				data-testid="info-card"
			>
				<div
					className="bg-white min-w-[600px] max-w-[768px] flex flex-col rounded-2xl shadow-2xl"
					ref={ref}
				>
					<div className="p-4 border-b border-border-primary">
						<h2>{header}</h2>
					</div>
					<div className="flex-1 p-4">
						<p>{message}</p>
					</div>
				</div>
			</div>
		);
	}
);
