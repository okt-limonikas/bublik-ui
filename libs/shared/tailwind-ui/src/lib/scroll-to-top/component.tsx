/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ComponentPropsWithRef, FC, forwardRef } from 'react';
import { AnimatePresence, motion, Variants } from 'framer-motion';

import { useScrollToTopPage } from '@/shared/hooks';

import { Icon } from '../icon';

const variants: Variants = {
	visible: { y: 0, opacity: 1 },
	hidden: { y: 72, opacity: 0 }
};

export interface ScrollToTopProps extends ComponentPropsWithRef<'button'> {
	offset?: number;
}

export const ScrollToTopPage: FC<ScrollToTopProps> = forwardRef(
	({ offset = 0 }, ref) => {
		const { isVisible, scrollToTop } = useScrollToTopPage(offset);

		return (
			<AnimatePresence>
				{isVisible && (
					<motion.div
						variants={variants}
						initial="hidden"
						animate="visible"
						exit="hidden"
						transition={{ type: 'just' }}
						className="fixed z-40 bottom-4 right-4"
					>
						<button
							className="flex items-center gap-1 p-3 text-white rounded-full bg-primary transition-all text-[1rem] font-medium leading-[0.875rem] border-2 border-transparent hover:border-[#94b0ff]"
							onClick={scrollToTop}
							data-testid="scroll-to-top"
							aria-label="Scroll to top"
							ref={ref}
						>
							<Icon name="ArrowShortTop" className="grid place-items-center" />
						</button>
					</motion.div>
				)}
			</AnimatePresence>
		);
	}
);
