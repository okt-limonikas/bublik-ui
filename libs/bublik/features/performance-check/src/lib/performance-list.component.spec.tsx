/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2024 OKTET LTD */
import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { PerformanceList } from './performance-list.component';

describe('<PerformanceList />', () => {
	it('should match snapshot', () => {
		const { asFragment } = render(
			<PerformanceList
				urls={[{ url: 'https://google.com', timeout: 1, label: 'Google' }]}
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
