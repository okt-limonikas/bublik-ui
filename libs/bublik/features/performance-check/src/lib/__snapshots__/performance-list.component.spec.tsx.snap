// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PerformanceList /> > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-2"
  >
    <h2
      class="text-xl font-semibold"
    >
      Health Check
    </h2>
    <ul
      class="flex flex-col gap-2"
    >
      <li
        class="flex items-center gap-2 justify-between border border-border-primary rounded px-4 py-2"
      >
        <a
          class="flex-1 flex-shrink-0 hover:underline text-primary"
          href="https://google.com"
          rel="noreferrer"
          target="_blank"
        >
          Google (1s)
        </a>
        <span
          class="tabular-nums w-[8ch] flex-1"
        >
          <div
            class="animate-pulse bg-gray-300 rounded-md h-[16px] w-[8ch] flex-1"
          />
        </span>
        <div
          class="w-3 h-3 rounded-full relative bg-gray-300"
        />
      </li>
    </ul>
    <button
      class="inline-flex items-center justify-center focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary transition-all appearance-none select-none disabled:shadow-[inset_0_0_0_1px_hsl(var(--colors-border-primary))] disabled:bg-white disabled:hover:bg-white disabled:text-text-menu px-3 text-[0.875rem] leading-[1.5rem] cursor-not-allowed pointer-events-none rounded-md hover:shadow-[inset_0_0_0_3px_#94b0ff] py-2 shadow-[inset_0_0_0_1px_hsl(var(--colors-border-primary))] bg-white hover:bg-white text-border-primary"
      disabled=""
    >
      Check
    </button>
  </div>
</DocumentFragment>
`;
